src/2dplane/GridStateSpace.cpp

```diff
@@ -16,6 +16,28 @@ bool GridStateSpace::stateValid(const Vector2f &pt) const {
     return PlaneStateSpace::stateValid(pt) && !_obstacleGrid.obstacleAt(_obstacleGrid.gridSquareForLocation(pt));
 }
 
+Vector2f GridStateSpace::intermediateState(const Vector2f &source, const Vector2f &target, float stepSize, float limit) const {
+    Vector2f delta = target - source;
+    delta = delta / delta.norm();   //  unit vector
+    if (stepSize == 0) {
+        stepSize = 0.1;
+    }
+
+    float n = _obstacleGrid.nearestObstacleDist(target);
+    if (stepSize > 50) {
```

Comment:
It would be best not to have any hard-coded constants like `50` is here.  Depending on what state space this RRT is being used in, a step size of 50 may be incredibly small or incredibly large.